src/plugins/action-queue/cleaners.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+//
+// Action Operations
+//
+import { type Cleaner, asArray, asEither, asNumber, asObject, asOptional, asString, asValue } from 'cleaners'
+
+import { type ActionEffect, type ActionOp, type ActionProgram, type ActionProgramState } from './types'
+
+const asError = (raw: any) => {
+  if (raw instanceof Error) return raw
+  throw new TypeError('Expected an Error')
+}
+
+export const asActionOp: Cleaner<ActionOp> = asEither(
+  asEither(
+    asObject({
+      type: asValue('seq'),
+      actions: asArray((raw: any) => asActionOp(raw))
+    }),
+    asObject({
+      type: asValue('par'),
+      actions: asArray((raw: any) => asActionOp(raw))
+    }),
+    asObject({
+      type: asValue('exchange:buy'),
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString),
+      exchangePluginId: asString
+    }),
+    asObject({
+      type: asValue('exchange:sell'),
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString),
+      exchangePluginId: asString
+    }),
+    asObject({
+      type: asValue('loan:borrow'),
+      borrowPluginId: asString,
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString)
+    })
+  ),
+  asEither(
+    asObject({
+      type: asValue('loan:deposit'),
+      borrowPluginId: asString,
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString)
+    }),
+    asObject({
+      type: asValue('loan:repay'),
+      borrowPluginId: asString,
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString)
+    }),
+    asObject({
+      type: asValue('loan:withdraw'),
+      borrowPluginId: asString,
+      nativeAmount: asString,
+      walletId: asString,
+      tokenId: asOptional(asString)
+    }),
+    asObject({
+      type: asValue('swap'),
+      fromWalletId: asString,
+      toWalletId: asString,
+      fromTokenId: asOptional(asString),
+      toTokenId: asOptional(asString),
+      nativeAmount: asString
```

Comment:
Is this a from amount of a to amount? In the core quote request, we have a direction flag to tell which way it is. We don't need to support both directions here (or maybe we could), but we do need to be clear about which one we are using.

Reply:
I went with adding an `amountFor: 'from' | 'to'` field.